prometheus/registry.go

```diff
@@ -505,22 +470,30 @@ func (r *registry) writePB(encoder expfmt.Encoder) error {
 		sort.Sort(metricSorter(mf.Metric))
 	}
 
-	// Write out MetricFamilies sorted by their name.
+	// Write out MetricFamilies sorted by their name, skipping those without
+	// metrics.
```

Comment:
The text parser deals with it correctly, if not we'd certainly have heard by now.